codebuild/bin/coverage_upload.sh

```diff
@@ -0,0 +1,24 @@
+#!/bin/bash
+# Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License").
+# You may not use this file except in compliance with the License.
+# A copy of the License is located at
+#
+#  http://aws.amazon.com/apache2.0
+#
+# or in the "license" file accompanying this file. This file is distributed
+# on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+# express or implied. See the License for the specific language governing
+# permissions and limitations under the License.
+#
+
+set -e
+
+# upload to codecov.io
+curl -Os https://uploader.codecov.io/latest/linux/codecov
```

Comment:
If codecov isn't accurate for our codebase, is this a useful step?

But I'm torn on whether it IS useful, from your description. The problem is that some single-line macros are very important and both branches SHOULD be tested. But some single-line macros aren't important and should be literally impossible to fail or test, like most of our extremely paranoid null checks.

Reply:
I've gone back and forth on this, but I've mostly arrived at "I can't fault CodeCov.io for pedantically accurate coverage". I _think_ it is still nice to have from the diff perspective, but don't have a strong opinion on that.

One other option that we could consider is modifying the `unit_test_coverage.info` to lie about our branch coverage, so that CodeCov.io reporting would be equivalent to the reporting at https://dx1inn44oyl7n.cloudfront.net/index.html

Reply:
I think my point is that the LLVM is arguably just as inaccurate as codecov.io, just in our favor. If I'm understanding the behavior you're describing, then codecove.io isn't being pedantic, it's being more accurate. We can't assume all single-line macros don't need to be tested any more than we can assume all single-line macros do need to be tested. We've got a weird mix.